streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoin.java

```diff
@@ -55,20 +61,22 @@ abstract class KStreamKStreamJoin<K, VLeft, VRight, VOut, VThis, VOther> impleme
     private final long windowsAfterMs;
 
     private final boolean outer;
-    private final Optional<String> outerJoinWindowName;
+    private final Optional<String> outerJoinWindowStoreName;
```

Comment:
nit: Do we really this additional Optional field? Seems we only used it in https://github.com/apache/kafka/pull/18111/files#diff-6ca18143cc0226e6d1e4d5180ff81596a72d53639ca5184bf1238350265382a6R121 which can be leveraged upon `outerJoinWindowStoreFactory` as well?